src/toast/scripts/toast_benchmark_satellite

```diff
@@ -0,0 +1,523 @@
+
+    # Log the config
+    if rank == 0:
+        if not os.path.isdir(args.out_dir):
+            os.makedirs(args.out_dir)
+        outlog = os.path.join(args.out_dir, "config_log.toml")
+        toast.config.dump_toml(outlog, config)
+
+    # selects appropriate case size
+    select_case(args, n_procs, n_nodes, avail_node_bytes, ops.binner.full_pointing, world_comm, log)
+
+    # Creates the focalplane file.
+    focalplane = make_focalplane(args, world_comm, log)
+
+    # from here on, we start the actual work (unless this is a dry run)
+    global_timer.start("toast_benchmark_satellite (science work)")
+    if args.dry_run is not None:
+        log.info_rank0("Exit from dry run.", world_comm)
+        # We are done!
+        sys.exit(0)
+
+    # Create a telescope for the simulation.
+    site = toast.instrument.SpaceSite("space")
+    telescope = toast.instrument.Telescope("satellite", focalplane=focalplane, site=site)
+
+    # Load the schedule file
+    schedule = make_schedule(args, world_comm, log)
+
+    # Find the group size for this job, either from command-line overrides or
+    # by estimating the data volume.
+    group_size = toast.job_group_size(
+        world_comm,
+        jobargs,
+        schedule=schedule,
+        focalplane=focalplane,
+        full_pointing=ops.binner.full_pointing,
+    )
+
+    # Create the toast communicator
+    comm = toast.Comm(world=world_comm, groupsize=group_size)
+
+    # Create the (initially empty) data
+    data = toast.Data(comm=comm)
+
+    # Simulate the telescope pointing
+    ops.sim_satellite.telescope = telescope
+    ops.sim_satellite.schedule = schedule
+    ops.sim_satellite.apply(data)
+
+    # Construct a "perfect" noise model just from the focalplane parameters
+    ops.default_model.apply(data)
+
+    # Set the pointing matrix operators to use the detector pointing
+    ops.pointing.detector_pointing = ops.det_pointing
+    ops.pointing_final.detector_pointing = ops.det_pointing
+
+    # Simulate sky signal from a map.
+    scan_map(args, rank, ops, data, log)
+
+    # Simulate detector noise
+    if ops.sim_noise.enabled:
+        ops.sim_noise.apply(data)
+
+    # Build up our map-making operation.
+    if ops.mapmaker.enabled:
+        run_mapmaker(ops, args, job.templates, data)
+
+    # end of the computations, sync and computes efficiency
+    global_timer.stop_all()
+    log.info_rank0("Gathering benchmarking metrics.", world_comm)
+    if world_comm is not None:
+        world_comm.barrier()
+    compute_science_metric(args, global_timer, n_nodes, rank, log)
+
+    # dumps all the timing information
+    timer = toast.timing.GlobalTimers.get()
+    timer.start("toast_benchmark_satellite (gathering and dumping timing info)")
+    alltimers = gather_timers(comm=world_comm)
+    if comm.world_rank == 0:
+        out = os.path.join(args.out_dir, "timing")
+        dump(alltimers, out)
+        with open(os.path.join(args.out_dir, "log"), "a") as f:
```

Comment:
We can remove this appending of the timing to the log file.  It is historical and actually assumes that the output of this script is being piped to a file called "log" in the output directory, which is not always the case.

Reply:
keeping line 500-501 I guess? (removing the log part but not the writing of timers to `timing`)